File: customer-service/src/main/java/com/wks/servicemarketplace/customerservice/config/ImmediateFeature.java
package com.wks.servicemarketplace.customerservice.config;

import org.glassfish.hk2.api.ServiceLocator;
import org.glassfish.hk2.utilities.ServiceLocatorUtilities;

import javax.inject.Inject;
import javax.ws.rs.core.Feature;
import javax.ws.rs.core.FeatureContext;

public class ImmediateFeature implements Feature {

    @Inject
    public ImmediateFeature(ServiceLocator locator) {
        ServiceLocatorUtilities.enableImmediateScope(locator);
    }

    @Override
    public boolean configure(FeatureContext context) {
        return true;
    }
}

File: auth-service/src/main/kotlin/com/wks/servicemarketplace/authservice/config/ImmediateFeature.kt
package com.wks.servicemarketplace.authservice.config

import org.glassfish.hk2.api.ServiceLocator
import org.glassfish.hk2.utilities.ServiceLocatorUtilities
import javax.inject.Inject
import javax.ws.rs.core.Feature
import javax.ws.rs.core.FeatureContext

class ImmediateFeature @Inject constructor(locator: ServiceLocator) : Feature {

    init {
        ServiceLocatorUtilities.enableImmediateScope(locator)
    }

    override fun configure(context: FeatureContext?) = true
}